channel.go

```diff
@@ -99,14 +99,15 @@ func newChannel(ctx context.Context, conn net.Conn) *Channel {
 
 // Run the Channel
 // in your application should create a new goroutine to loop run this function
-func (channel *Channel) Run() {
+func (channel *Channel) Run(stop <-chan struct{}) error {
```

Comment:
But I think that Run should be an empty args, maybe you could keep the channel in the Channel struct? right ?

Reply:
Try do a deep think if you could pass a channel from Run sematic function ?